src/main/java/DukePackage/Storage.java

```diff
@@ -0,0 +1,224 @@
+package DukePackage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.BufferedWriter;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.ArrayList;
+
+
+/**
+ * The Storage class is responsible for managing the storage of tasks in a task list.
+ */
+public class Storage {
+
+    /**
+     * The list of tasks.
+     */
+    protected ArrayList<Task> taskList;
+    protected final String MATCHING_STRING = "     Here are the matching tasks in your list:";
+
+    /**
+     * Constructs a Storage object with an empty task list.
+     */
+    public Storage() {
+        this.taskList = new ArrayList<Task>();
+    }
+
+    /**
+     * Retrieves the task list.
+     *
+     * @return The task list.
+     */
+    public ArrayList<Task> getTaskList() {
+        return this.taskList;
+    }
+
+    /**
+     * Adds a task to the task list.
+     *
+     * @param t The task to be added.
+     */
+    public void addList(Task t) {
+        this.taskList.add(t);
+    }
+
+    /**
+     * Deletes a task from the task list.
+     *
+     * @param id The index of the task to be deleted.
+     */
+    public void deleteTask(int id) {
+        System.out.println("     Noted. I've removed this task:");
+        Task t = this.taskList.get(id);
+        t.printMarking(false);
+        this.taskList.remove(id);
+        int size = this.taskList.size();
+        System.out.printf("\n     Now you have %d tasks in the list.\n", size);
+    }
+
+    /**
+     * Writes the task list to a file.
+     */
+    public void writeTasksToFile() {
```

Comment:
This method is currently pretty long, would it be possible to compress its implementations into fewer lines by further abstracting out the different parts of it such as the for-loop?